crates/workspace/src/toml_options.rs

```diff
@@ -0,0 +1,121 @@
+use crate::settings::FormatSettings;
+use crate::settings::IndentStyle;
+use crate::settings::IndentWidth;
+use crate::settings::LineEnding;
+use crate::settings::LineLength;
+use crate::settings::MagicLineBreak;
+use crate::settings::Settings;
+
+/// The Rust representation of `air.toml`
+///
+/// The names and types of the fields in this struct determine the names and types
+/// that can be specified in the `air.toml`.
+///
+/// Every field is optional at this point, nothing is "finalized".
+/// Finalization is done in [TomlOptions::into_settings].
+///
+/// Global options are specified at top level in the TOML file.
+/// All other options are nested within their own `[table]`.
+#[derive(Clone, Debug, PartialEq, Eq, Default, serde::Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct TomlOptions {
+    /// Global options affecting multiple commands.
+    #[serde(flatten)]
+    pub global: GlobalTomlOptions,
+
+    /// Options to configure code formatting.
+    pub format: Option<FormatTomlOptions>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Default, serde::Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct GlobalTomlOptions {
+    /// The line length at which the formatter prefers to wrap lines.
+    ///
+    /// The value must be greater than or equal to `1` and less than or equal to `320`.
+    ///
+    /// Note: While the formatter will attempt to format lines such that they remain
+    /// within the `line-length`, it isn't a hard upper bound, and formatted lines may
+    /// exceed the `line-length`.
+    pub line_length: Option<LineLength>,
+
+    /// The number of spaces per indentation level (tab).
+    ///
+    /// The value must be greater than or equal to `1` and less than or equal to `24`.
+    ///
+    /// Used by the formatter to determine the visual width of a tab.
+    ///
+    /// This option changes the number of spaces the formatter inserts when
+    /// using `indent-style = "space"`. It also represents the width of a tab when
+    /// `indent-style = "tab"` for the purposes of computing the `line-length`.
+    pub indent_width: Option<IndentWidth>,
```

Comment:
Both `line_length` and `indent_width` are top-level global options used by "all of air" because if we consider a linter then it will need access to these options to determine if a line exceeds the line length.

Compare that with `indent_style`, `line_ending`, and `ignore_magic_line_break` which are more clearly format specific and live under the `[format]` table header.